src/kv/untyped_tx_view.h

```diff
@@ -12,12 +12,37 @@ namespace kv::untyped
   using SerialisedKeyHasher = std::hash<SerialisedEntry>;
 
   using VersionV = kv::VersionV<SerialisedEntry>;
-  using State =
-    kv::State<SerialisedEntry, SerialisedEntry, SerialisedKeyHasher>;
+
+  struct k_size
+  {
+    uint32_t operator()(const SerialisedEntry& key)
+    {
+      return sizeof(uint64_t) + key.size();
```

Comment:
This fixes the _size_ calculation logic at type-definition, but the actual serialisation is still an arbitrary function - there's a stronger connection between these 2 functors, so they should be kept together. If all we're really doing is calling `serialized::write<T>`, we could add `serialized::size<T>` to return the required size?

And what's the motivation for making this snapshot-serialisation pluggable - can we just call `serialized::` directly in `champ_map.h` rather than a templated or member functor?

Reply:
discussed offline. We came to a different solution. All the serialization, deserialization, size, ... are now in a separate file. We should look at this again before pushing the PR.